cmd/internal/planetscale_edge_database.go

```diff
@@ -314,9 +318,13 @@ func (p PlanetScaleEdgeDatabase) sync(ctx context.Context, tc *psdbconnect.Table
 			}
 		}
 
-		if watchForVgGtidChange && tc.Position != stopPosition {
+		if (watchForVgGtidChange && tc.Position != stopPosition) || (lastRestCount == 0 && len(res.Result) == 0 && tc.Position == stopPosition) {
```

Comment:
am i reading this right: stop if there are two empty results in a row and we've reached the stop position?

Reply:
that's the idea. In testing, for a database with data and no active writes, this tended to be indicative of the stream being caught up. 

The existing behavior will just wait for a timeout in that situation, which seems unwise if we are going to increase the timeout.